redskyctl/internal/commands/experiments/experiments.go

```diff
@@ -101,47 +105,100 @@ func (n *name) experimentName() experimentsv1alpha1.ExperimentName {
 	return experimentsv1alpha1.NewExperimentName(n.Name)
 }
 
-// numberSuffixPattern matches the trailing digits, for example the number on the end of a trial name
-var numberSuffixPattern = regexp.MustCompile(`(.*?)(?:[/\-]([[:digit:]]+))?$`)
+// trialNumber returns the trial number extracted from the name, values less then zero indicate no
+// number is present.
+func (n *name) trialNumber() int64 {
+	return n.Number
+}
 
 // parseNames parses a list of arguments into structured names
 func parseNames(args []string) ([]name, error) {
-	var t resourceType
 	names := make([]name, 0, len(args))
+
+	var defaultType string
 	for _, arg := range args {
-		n := name{Type: t, Name: arg, Number: -1}
-		if sm := numberSuffixPattern.FindStringSubmatch(n.Name); sm != nil && sm[2] != "" {
-			n.Number, _ = strconv.ParseInt(sm[2], 10, 64)
-			n.Name = sm[1]
+		// Split the argument into type/name/number
+		argType, argName, argNumber := splitArg(arg, defaultType)
+		argIsTypeOnly := argName == "" && argNumber == ""
+
+		// Normalize the type
+		normalType, pluralType, err := normalizeType(argType)
+		if err != nil {
+			return nil, err
+		}
+
+		// Set the default type if it hasn't been set and no name was supplied
+		if defaultType == "" && argIsTypeOnly {
+			defaultType = pluralType
+			continue
+		}
+
+		// Create a new name
+		n := name{Type: normalType, Name: argName, Number: -1}
+
+		// Special case where trial can alternatively end with "-<NUM>" instead of "/<NUM>"
+		if n.Type == typeTrial && argNumber == "" && pluralType == string(normalType) {
+			if pos := strings.LastIndex(argName, "-"); pos > 0 {
+				n.Name, argNumber = argName[0:pos], argName[pos+1:]
+			}
 		}
 
-		p := strings.SplitN(n.Name, "/", 2)
-		if len(p) > 1 || t == "" {
-			nt, err := normalizeType(p[0])
+		// Parse the number, if present
+		if argNumber != "" {
+			if n.Type != typeTrial {
+				return nil, fmt.Errorf("%s name cannot include a number: %s", n.Type, arg)
```

Comment:
just to make sure I'm on the same page; this isn't a new limitation correct?
`experiment-123` is still a valid experiment name, but `experiment/123` would not be?